### DISCUSSION TITLE
GitHub copilot cannot work correctly on vscode remote server

### DISCUSSION THREAD

**Discussion by SimpleXiaohu** [community/community#14907] [Copilot Conversations] [answered]
I want to use github copilot on my remote server, but it doesn't work. And I get info on copilot output :
```
[INFO] [ghostText] [2022-04-14T02:02:57.110Z] Cancelled after extractPrompt
[INFO] [ghostText] [2022-04-14T02:02:57.110Z] Cancelled after extractPrompt
[INFO] [ghostText] [2022-04-14T02:02:57.112Z] Cancelled after extractPrompt
[INFO] [ghostText] [2022-04-14T02:02:57.112Z] Cancelled after extractPrompt
[INFO] [ghostText] [2022-04-14T02:05:07.983Z] Cancelled after extractPrompt
[INFO] [ghostText] [2022-04-14T02:05:07.984Z] Cancelled after extractPrompt
[INFO] [ghostText] [2022-04-14T02:09:26.029Z] Cancelled after requestMultiline
[INFO] [ghostText] [2022-04-14T02:09:26.029Z] Cancelled after requestMultiline
[INFO] [ghostText] [2022-04-14T02:09:26.029Z] Cancelled after requestMultiline
```
How to fix it?

**Comment by SimpleXiaohu** [comment, marked as answer]
Finally, I find that it is because of the  bad network 😢.

**Comment by LinqLover** [comment]
Same issue here, same setup, but stable 50 Mbit/s. Reloading the window/updating VS Code did not help. Any one else experienced the problem and knows how to solve it?

<details><summary>Details</summary><pre><code>[INFO] [ghostText] [2022-07-18T18:30:52.236Z] Cancelled after extractPrompt
[INFO] [ghostText] [2022-07-18T18:30:52.236Z] Cancelled after extractPrompt
[INFO] [ghostText] [2022-07-18T18:30:53.468Z] Cancelled after requestMultiline
[INFO] [default] [2022-07-18T18:30:56.231Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-experiment-b-centralus
[INFO] [ghostText] [2022-07-18T18:31:28.986Z] Cancelled after extractPrompt
[INFO] [ghostText] [2022-07-18T18:31:31.123Z] Cancelled after requestMultiline
[INFO] [ghostText] [2022-07-18T18:31:31.124Z] Cancelled after extractPrompt
[INFO] [ghostText] [2022-07-18T18:31:31.124Z] Cancelled after extractPrompt
[INFO] [ghostText] [2022-07-18T18:31:35.760Z] Cancelled after requestMultiline
[INFO] [ghostText] [2022-07-18T18:31:37.756Z] Cancelled after requestMultiline
[INFO] [ghostText] [2022-07-18T18:34:56.143Z] Cancelled after extractPrompt
[INFO] [ghostText] [2022-07-18T18:36:54.059Z] Cancelled after requestMultiline
[INFO] [default] [2022-07-18T18:36:55.166Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-experiment-b-centralus
[INFO] [default] [2022-07-18T18:36:55.679Z] request.response: [https://copilot-proxy.githubusercontent.com/v1/engines/copilot-experiment-b-centralus/completions] took 512 ms
[INFO] [streamChoices] [2022-07-18T18:36:55.685Z] solution 0 returned. finish reason: ["stop"] finishOffset: [undefined] completionId: [{cmpl-5VPtvW7i3mHmG72l4ZDTzsSnQFmXK}] created: [{1658169415}]
[INFO] [ghostText] [2022-07-18T18:36:12.230Z] Cancelled after extractPrompt
[ERROR] [default] [2022-07-18T18:36:17.323Z] Request Error: Client network socket disconnected before secure TLS connection was established
[ERROR] [default] [2022-07-18T18:36:22.620Z] Request Error: Client network socket disconnected before secure TLS connection was established
</code></pre></details>

**Comment by shwq774791378** [comment]
> 14T02

sdk/keyvault/azure-security-keyvault-secrets/src/samples/java/com/azure/security/keyvault/secrets/IdentityReadmeSamples.java

**Reply by Bittrex11** [reply]
Bittrex11

**Comment by pablodz** [comment]
It's about packet loss, 5% will generate this problem
